File: Easy/27GenerateDocument.py
def generateDocument(characters, document):
    # Write your code here.
    characterCounts = {}
    for character in characters:
        if character not in characterCounts:
            characterCounts[character] = 0
        characterCounts[character] += 1

    for character in document:
        if character not in characterCounts or characterCounts[character] == 0:
            return False
        characterCounts[character] -= 1

    return True


print(generateDocument("Bste!hetsi ogEAxpelrt x ", "AlgoExpert is the Best!"))

File: Easy/27GenerateDocument.js
function generateDocument(characters, document) {
  // Write your code here.
  let characterCounts = {};
  for (let character of characters) {
    if (!characterCounts.hasOwnProperty(character)) {
      characterCounts[character] = 0;
    }
    characterCounts[character] += 1;
  }
  for (let character of document) {
    if (
      !characterCounts.hasOwnProperty(character) ||
      characterCounts[character] == 0
    ) {
      return false;
    }
    characterCounts[character] -= 1;
  }
  return true;
}

console.log("====================================");
console.log(
  generateDocument("Bste!hetsi ogEAxpelrt x ", "AlgoExpert is the Best!")
);
console.log("====================================");
